packages/core/src/services/ai/index.ts

```diff
@@ -137,23 +138,26 @@ export async function ai({
     const messages = rule.messages
     const model = config.model
     const tools = config.tools
-    const llmProvider = createProvider({
+    const llmProviderResult = createProvider({
       messages,
       provider: apiProvider,
       apiKey,
       url: url ?? undefined,
       config,
     })
 
-    if (llmProvider.error) return llmProvider
+    if (llmProviderResult.error) return llmProviderResult
 
     const languageModel = customLanguageModel
       ? customLanguageModel
-      : llmProvider.value(model, {
-          cacheControl: config.cacheControl ?? false,
-          // Propagate provider config options for this language model
-          ...config,
-        })
+      : apiProvider.provider === Providers.Anthropic
+        ? (llmProviderResult.value as AnthropicProvider)(model, {
```

Comment:
@andresgutgon I'm assuming this cachecontrol header was only intended for anthropic provider?

Reply:
I think is for anthropic and all anthropic providers like Vertex and Bedrock